### PR TITLE
fix id for title.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh 

use github rule: https://github.com/jch/html-pipeline/blob/master/lib/html/pipeline/toc_filter.rb

### CODE REVIEW

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHeadingBlockRule.cs [hunk 1]
```diff
@@ -27,7 +27,7 @@ public virtual IMarkdownToken TryMatch(IMarkdownParser parser, IMarkdownParsingC
                     t.Rule,
                     t.Context,
                     p.TokenizeInline(t.SourceInfo.Copy(match.Groups[2].Value)),
-                    Regex.Replace(match.Groups[2].Value.ToLower(), @"[^\p{L}\p{N}]+", "-").Trim('-'),
+                    Regex.Replace(match.Groups[2].Value.ToLower(), @"[^\p{L}\p{N}\- ]+", "").Replace(' ', '-'),
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.MarkdownLite/Basic/BlockRules/MarkdownHeadingBlockRule.cs]: >Replace(' ', '-') [](start = 93, length = 17)

need Trim('-')?

**Comment by ansyral** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:+1:

**Pull request merged**